[
   {
      "author_association" : "MEMBER",
      "body" : "Meh. I think it's reasonable to expect failures if the user is manually modifying their wallet file by doing direct database operations themselves. Since it is completely unexpected to have legacy and descriptor wallet things mixed, I don't think we should just silently ignore such issues and instead return an error.",
      "created_at" : "2022-11-06T15:19:34Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/26462#issuecomment-1304824794",
      "id" : 1304824794,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/26462",
      "node_id" : "IC_kwDOABII585Nxgva",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1304824794/reactions"
      },
      "updated_at" : "2022-11-06T15:19:34Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1304824794",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/3782274?v=4",
         "events_url" : "https://api.github.com/users/achow101/events{/privacy}",
         "followers_url" : "https://api.github.com/users/achow101/followers",
         "following_url" : "https://api.github.com/users/achow101/following{/other_user}",
         "gists_url" : "https://api.github.com/users/achow101/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/achow101",
         "id" : 3782274,
         "login" : "achow101",
         "node_id" : "MDQ6VXNlcjM3ODIyNzQ=",
         "organizations_url" : "https://api.github.com/users/achow101/orgs",
         "received_events_url" : "https://api.github.com/users/achow101/received_events",
         "repos_url" : "https://api.github.com/users/achow101/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/achow101/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/achow101/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/achow101"
      }
   },
   {
      "author_association" : "CONTRIBUTOR",
      "body" : "> Meh. I think it's reasonable to expect failures if the user is manually modifying their wallet file by doing direct database operations themselves. Since it is completely unexpected to have legacy and descriptor wallet things mixed, I don't think we should just silently ignore such issues and instead return an error.\r\n\r\nRight, throwing an error seems more reasonable. Obviously it wouldn't be the user itself modifying an existing wallet, but e.g. an external malicious actor crafting and handing out these mixed wallets, hoping that users would open them to cause node crashes.\r\n\r\nThe PR in its current state is just the easiest \"avoid crash\"-remedy. Will look into your proposed\r\nsolution in a bit, this probably needs a new `DBErrors` category.",
      "created_at" : "2022-11-06T17:35:56Z",
      "html_url" : "https://github.com/bitcoin/bitcoin/pull/26462#issuecomment-1304853150",
      "id" : 1304853150,
      "issue_url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/26462",
      "node_id" : "IC_kwDOABII585Nxnqe",
      "performed_via_github_app" : null,
      "reactions" : {
         "+1" : 0,
         "-1" : 0,
         "confused" : 0,
         "eyes" : 0,
         "heart" : 0,
         "hooray" : 0,
         "laugh" : 0,
         "rocket" : 0,
         "total_count" : 0,
         "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1304853150/reactions"
      },
      "updated_at" : "2022-11-06T17:35:56Z",
      "url" : "https://api.github.com/repos/bitcoin/bitcoin/issues/comments/1304853150",
      "user" : {
         "avatar_url" : "https://avatars.githubusercontent.com/u/91535?v=4",
         "events_url" : "https://api.github.com/users/theStack/events{/privacy}",
         "followers_url" : "https://api.github.com/users/theStack/followers",
         "following_url" : "https://api.github.com/users/theStack/following{/other_user}",
         "gists_url" : "https://api.github.com/users/theStack/gists{/gist_id}",
         "gravatar_id" : "",
         "html_url" : "https://github.com/theStack",
         "id" : 91535,
         "login" : "theStack",
         "node_id" : "MDQ6VXNlcjkxNTM1",
         "organizations_url" : "https://api.github.com/users/theStack/orgs",
         "received_events_url" : "https://api.github.com/users/theStack/received_events",
         "repos_url" : "https://api.github.com/users/theStack/repos",
         "site_admin" : false,
         "starred_url" : "https://api.github.com/users/theStack/starred{/owner}{/repo}",
         "subscriptions_url" : "https://api.github.com/users/theStack/subscriptions",
         "type" : "User",
         "url" : "https://api.github.com/users/theStack"
      }
   }
]
